doc/development/twister.md

```diff
@@ -1,59 +1,281 @@
+
+Many more field are described in the [Tests] documentation.
+
+[Tests]: https://docs.zephyrproject.org/latest/develop/test/twister.html#tests
+[Selecting platform scope]:
+  https://docs.zephyrproject.org/latest/develop/test/twister.html#selecting-platform-scope
+
+## Twister's command line options
+
+Twister has a lot of command line options that can be used to control its behavior. Run
+`east twister --extra-help` to see them.
+
+Some more important options are:
+
+- `-T TESTSUITE_ROOT, --testsuite-root TESTSUITE_ROOT` - Base directory to recursively search for
+  test cases. All `testcase.yaml` and `sample.yaml` files under here will be processed. May be
+  called multiple times. Defaults to the `samples/` and `tests/` directories at the base of the
+- `-p PLATFORM, --platform PLATFORM` - Platform filter for testing. This option may be used multiple
+  times. Test suites will only be built/run on the platforms specified. If this option is not used,
+  then platforms marked as default in the platform metadata file will be chosen to build and test.
+  Zephyr tree, so make sure that you always specify it.
+- `-t TAG, --tag TAG` - Specify tags to restrict which tests to run by tag value. Default is to not
+  do any tag filtering. Multiple invocations are treated as a logical 'or' relationship.
+- `-b, --build-only` - Only build the code, do not attempt to run the code on targets.
+- `-f, --only-failed` - Run only those tests that failed the previous twister run invocation.
+- `-y, --dry-run` - Create the filtered list of test cases, but don't actually run them. Useful if
+  you're just interested in the test plan generated for every run and saved in the specified output
+  directory (`testplan.json`).
+- `--overflow-as-errors` - Treat FLASH/RAM/SRAM overflows as errors. **This one is important. If not
+  used the test that overflows is considered to be skipped and not that it has failed.**
+- `-v, --verbose` - Emit build logs, call multiple times to increase verbosity.
+
+## `twister-out` folder
+
+With every invocation of the `east twister` a `twister-out` folder is created. It contains all build
+folders of the created projects, as well as the report files.
+
+Keep in mind, if you keep running `east twister` command and not deleting the `twister-out` folder,
+then the command will rename it to `twister-out.X` (Where `X` is a unique ascending number, starting
+with 1) before running the full build.
+
+## Cookbook style examples
 
-Due to the number of features that it has it can take some time to figure out how to do something
-that you want, see `east twister --extra-help` for the supported options.
+### Running tests on the Host PC
 
-Also, see [Zephyr's documentation](https://docs.zephyrproject.org/latest/develop/test/twister.html)
-for more info.
+You want to build and run all tests under `tests/` directory on the Host PC.
+
+`testcase.yaml` file in the all projects under `tests/` directory should look like this:
```

Comment:
Reading this it kind of sounds like all tests must have `platform_allow: native_sim`, but this is not true.

The command will run all tests that have `platform_allow: native_sim`.
You should reword this.



```suggestion
You want to build and run all tests under the `tests/` directory meant for native testing.

`testcase.yaml` file in all the applicable projects under the `tests/` directory should contain this:
```

Reply:
I have kept the Host PC part, reworded rest.